test/integration.test.js

```diff
@@ -3,25 +3,38 @@
  */
 
 const { readFile } = require('fs').promises;
+const { existsSync } = require('fs');
 const path = require('path');
 const Generator = require('../lib/generator');
 const dummySpecPath = path.resolve(__dirname, './docs/dummy.yml');
 const refSpecPath = path.resolve(__dirname, './docs/apiwithref.json');
 const refSpecFolder = path.resolve(__dirname, './docs/');
 const crypto = require('crypto');
+const {mkdirSync} = require('fs');
+const {exists, writeFile} = require('../lib/utils');
 const mainTestResultPath = 'test/temp/integrationTestResult';
 const reactTemplate = 'test/test-templates/react-template';
 const nunjucksTemplate = 'test/test-templates/nunjucks-template';
+const logMessage = require('../lib/logMessages.js');
+const log = require('loglevel');
 
 describe('Integration testing generateFromFile() to make sure the result of the generation is not changend comparing to snapshot', () => {
   const generateFolderName = () => {
-    //you always want to generate to new directory to make sure test runs in clear environment
+    // you always want to generate to new directory to make sure test runs in clear environment
+    // normalize the path for cross-platform compatibility
+    // return path.normalize(path.resolve(mainTestResultPath, crypto.randomBytes(4).toString('hex')));
     return path.resolve(mainTestResultPath, crypto.randomBytes(4).toString('hex'));
   };
 
   jest.setTimeout(60000);
   const testOutputFile = 'test-file.md';
 
+  beforeAll(() => {
+    if (!existsSync(path.join(reactTemplate, 'package.json'))) {
```

Comment:
why we need this?

Reply:
Deleted. I used it during development because I could not find the package.json sometimes.